go/cmd/dolt/commands/init.go

```diff
@@ -154,7 +155,15 @@ func (cmd InitCmd) Exec(ctx context.Context, commandStr string, args []string, d
 	}
 
 	requiresFunHash := apr.Contains(funHashFlag)
-	err := dEnv.InitRepoWithTime(context.Background(), types.Format_Default, name, email, initBranch, t, requiresFunHash)
+	commitMeta := func() datas.CommitMetaGenerator {
+		if requiresFunHash {
+			return datas.MakeFunCommitMetaGenerator(name, email, t)
+		} else {
+			return datas.MakeCommitMetaGenerator(name, email, t)
+		}
+	}()
```

Comment:
This could be simplified a bit... no need to create the anonymous function just to call it immediately...

```suggestion
	generator := datas.MakeCommitMetaGenerator(name, email, t)
	if requiresFunHash {
		generator = datas.MakeFunCommitMetaGenerator(name, email, t)
	}
```

Reply:
I did the anonymous function because it avoids calling `MakeCommitMetaGenerator` if it doesn't have to. In practice, this code will only run once and the performance impact is so minor that it really doesn't matter. If you feel like your suggestion is more readable, then lets go with that.